packages/protocol/contracts/L1/libs/LibVerifying.sol

```diff
@@ -153,22 +153,35 @@ library LibVerifying {
     }
 
     function _rewardProvers(
+        TaikoData.Config memory config,
         TaikoData.ForkChoice storage fc,
         uint256 reward,
         TkoToken tkoToken
     ) private {
-        uint sum = 2 ** fc.provers.length - 1;
-        for (uint i = 0; i < fc.provers.length; ++i) {
-            uint weight = (1 << (fc.provers.length - i - 1));
+        uint start;
+        uint count = fc.provers.length;
+
+        if (config.enableSpecialFirstProver) {
+            start = 1;
+            count -= 1;
+        }
+
+        uint sum = (1 << count) - 1;
+        uint weight = 1 << (count - 1);
+        for (uint i = 0; i < count; ++i) {
             uint proverReward = (reward * weight) / sum;
```

Comment:
Shall we also skip updating the `state.avgProofTime` when the proof is submitted by the `oracle prover`? seems this state may also affect the reward value and the proof deadline for other decentralized provers.